packages/admin-panel/src/VizBuilderApp/App.js

```diff
@@ -37,12 +37,8 @@ export const App = ({ Navbar, Footer }) => {
       <Container>
         {Navbar && <Navbar user={user} isBESAdmin={isBESAdmin} />}
         <Switch>
-          <Route path="/viz-builder/new" exact>
-            <CreateNew />
-          </Route>
-          <Route path="/">
-            <Main />
-          </Route>
+          <Route path="/viz-builder/new" exact component={CreateNew} />
+          <Route path="/viz-builder/:visualisationId?" component={Main} />
```

Comment:
Any reason to add the components using the component prop? It's only a small thing but [the convention in react-router ](https://reactrouter.com/web/api/Route/route-render-methods) is to pass them as children.

Reply:
Ah the main reason is only by this way, the `Main` component can access the `:visualisationId` parameter from the Route path (I tried some other ways but was not successful)

Reply:
You should be able to just use [the react-router `useParams`](https://reactrouter.com/web/api/Hooks/useparams) hook to get the param in the `Main` component. 
eg. ` const { visualisationId } = useParams();` 

Then you wont need to add the components as named props any more on the `Route` above. Let me know if that doesn't work.

Reply:
Thanks for that suggestion, it works! That's much better